include/meshio/utilities.hpp

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2017, Lakshman Anumolu, Pradeep Garigipati
+ * All rights reserved.
+ *
+ * This file is part of MeshIO whose distribution is governed by
+ * the BSD 2-Clause License contained in the accompanying LICENSE.txt
+ * file.
+ */
+
+#ifndef MESHIO_UTILITIES_HPP_
+#define MESHIO_UTILITIES_HPP_
+
+#include <string>
+#include <cmath>
+
+namespace meshio
+{
+
+/**
+ * Extracts filename from path.
+ */
+static inline std::string getFileBaseName(const std::string& pFileName)
```

Comment:
Shouldn't this return the last word found in the path? Even that is a valid file name. All files doesn't need to have an extension necessarily. If you intend to have this function not handle such cases, mention that in function documentation to be clear for whoever might use this in future. May be in a section such as `limitations` in the function docs.